sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -304,7 +304,10 @@ object HiveTypeCoercion {
   }
 
   /**
-   * Convert all expressions in in() list to the left operator type
+   * Convert the value and in list expressions to the common operator type
+   * by looking at all the argument types and finding the closest one that
+   * all the arguments can be cast to. When no common operator type is found
+   * an Analysis Exception is raised.
```

Comment:
@cloud-fan 
You are right Wenchen.I just wanted to somehow mention that an exception will be raised on a data type mistmatch. I will reword it to the way you suggest. ""the original one will be returned and an Analysis Exception will be raised at type checking phase". Let me push a commit.